### PR TITLE
Metrics HOF

### PR DESCRIPTION
**Pull request by Velenir**
Turns `executeMetrics` into a higher-order-function, as promised [here](https://github.com/gnosis/dex-price-estimator/pull/26#discussion_r414442491)
That way metrics logic is separate from routes and OrderBook logic

### CODE REVIEW

**Review by anxolin — APPROVED** [review]
🎩Very elegant 🎩

**Comment by anxolin** [comment]
Please, make sure Travis is happy https://travis-ci.org/github/gnosis/dex-price-estimator/builds/680040726

I also got one time the test failing. Re-running travis solved the issue. I think that they depend on the node, so they can behave a bit flaky

File: src/utilities.ts [hunk 1]
```diff
@@ -52,28 +53,35 @@ interface ExecuteWithMetricsParams<T extends string> {
...
+          endTimer(labelValues)
+        }
+      }
     }
-  }
 }
+
+export const withOrderBookMetrics = withMetrics(orderBookMetrics)
+export const withBuyAmountEstimationMetrics = withMetrics(buyAmountEstimationMetrics)
+export const withOrderBookFetcherMetrics = withMetrics(orderBookFetcher)
```
Comment (alfetopito) [inline src/utilities.ts]: I think this doesn't belong here.

HOF function, yes, that is a utility.
But the actual function combined with the metrics should either be exported from where the metrics is defined or where it's used.

**Review by alfetopito — APPROVED** [review]
:level_slider: :up:

**Pull request merged**